### ISSUE TITLE
false positive: some builtin methods are incorrectly reported as private

### ISSUE THREAD

**Issue by Qwinci** [intellij-rust/intellij-rust#10567]
## Environment

* **IntelliJ Rust plugin version:** 0.4.197.5406-231-nightly
* **Rust toolchain version:** 1.72.0-nightly (37998ab50 2023-06-11) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2023.1.3 (CL-231.9011.31)
* **Operating system:** Linux 6.1.31-gentoo
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.emulate.terminal

## Problem description
CLion incorrectly reports some builtin methods as private.

## Steps to reproduce

![image](https://github.com/intellij-rust/intellij-rust/assets/32550582/117000e9-00e6-4fa6-9b3b-fb69866f396d)

**Comment by Qwinci** [comment]
Also happens on the latest stable IntelliJ Rust and Rust nightly-2023-05-01

**Comment by Maksim20023** [comment]
Hi! Thank you for reporting. 
This issue is related to a false positive trigger of "[Method `wrapping_add` is private [E0624]](https://doc.rust-lang.org/error_codes/E0624.html)". We'll investigate the reasons for such behavior and inform you in reply when we find a solution. If you have additional information, feel free to share with us

**Comment by blueglyph** [comment]
I noticed that I had the problem on signed integers but not unsigned integers:

![image](https://github.com/intellij-rust/intellij-rust/assets/31706195/ba8a81ff-09f4-4f5d-a568-63c9cf17cb70)

The `i32` and `u64` implementations are done differently:
* `u64` (like all unsigned ints) is an implementation on the primitive (`impl u64`), which includes a declarative macro
* `i32` (like all signed ints) is a declarative macro doing a trait implementation for the type (`impl Int for $uty`)

In case it helps, this code shows the difference:

```rust
#[test]
pub fn test_ints() {
    let mut a: u64 = 10;
    let mut b: i32 = 10;
    a = a.wrapping_mul(10);
    b = b.wrapping_mul(10);
    assert_eq!(a, 100);
    assert_eq!(b, 100);
}
```

## Environment

* **IntelliJ Rust plugin version:** 0.4.197.5401-231
* **Rust toolchain version:** 1.69.0 (84c898d65 2023-04-16) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2023.1.3 Ultimate Edition (IU-231.9161.38)
* **Operating system:** Windows 10.0
* **Macro expansion:** enabled
